website/app/components/doc/layout/index.js

```diff
@@ -0,0 +1,87 @@
+import Component from '@glimmer/component';
+import { htmlSafe } from '@ember/template';
+import { assert } from '@ember/debug';
+
+export const DIRECTIONS = ['horizontal', 'vertical'];
+export const ALIGNMENTS = ['left', 'right', 'center', 'justify'];
+
+// sanitize & validate custom spacing value:
+const cssUnitRegEx = /^-?((\d+)|(\d+\.\d+)|(\.\d+))(px|rem|em|%)$/;
+
+export default class DocLayoutIndexComponent extends Component {
+  /**
+   * Get the direction for the component
+   * Accepted values: 'horizontal', 'vertical'
+   * @param direction
+   * @type {string}
+   * @default 'horizontal'
+   */
+  get direction() {
+    let { direction = 'horizontal' } = this.args;
+
+    assert(
+      '@direction for "Doc::Layout" must have a valid value',
+      DIRECTIONS.includes(direction)
+    );
+
+    return direction;
+  }
+
+  /**
+   * Get the inline custom spacing style.
+   * Accepted values: '1px', '2.4rem', '3em', '2%', etc.
+   * @param spacing
+   * @type {string} The "style" attribute to apply to the item.
+   */
+  get spacing() {
+    let { spacing } = this.args;
+
+    assert(
+      '@spacing for "Doc::Layout" must have a valid value',
+      spacing === undefined || spacing.match(cssUnitRegEx)
+    );
+
+    if (spacing !== undefined) {
+      if (spacing.match(cssUnitRegEx)) {
+        return htmlSafe(`--doc-layout-spacing: ${spacing}`);
+      }
+    }
```

Comment:
Code can be simplified here:
- the `{{style}}` helper already validates the spacing unit (if it's invalid, is simply ignored)
- instead of passing a CSS variable and then use it to define a `gap` you can directly pass the `gap` CSS declaration

```suggestion
    if (spacing !== undefined) {
      return htmlSafe(`gap: ${spacing}`);
    }
```

Reply:
Thanks, this doesn't validate the passed value this way. I'll see if there's any other simplifications I could make.